cmds/modules/migrate/main.go

```diff
@@ -0,0 +1,218 @@
+	Action: action,
+}
+
+type Farm struct {
+	ID      uint64 `json:"farmID"`
+	Name    string `json:"name"`
+	Address string `json:"stellarAddress"`
+}
+
+// farmsWithWallet returns v3 farms that has the given pay address
+func farmsWithWallet(ctx context.Context, url, wallet string) ([]Farm, error) {
+	cl := graphql.NewClient(url)
+
+	req := graphql.NewRequest(`
+    query ($address: String!) {
+		farms(where: {stellarAddress_eq: $address}) {
+		  name
+		  farmId
+		  stellarAddress
+		}
+	  }
+
+	`)
+
+	req.Var("address", wallet)
+	var response struct {
+		Farms []Farm `json:"farms"`
+	}
+
+	if err := cl.Run(ctx, req, &response); err != nil {
+		return nil, errors.Wrap(err, "failed to query farms with payment address")
+	}
+
+	return response.Farms, nil
+}
+
+func walletAddress(farm *directory.Farm) string {
+	const ASSET = "TFT"
+
+	for _, addr := range farm.WalletAddresses {
+		if addr.Asset == ASSET {
+			return addr.Address
+		}
+	}
+
+	return ""
+}
+
+func migrationFarm(candidates []Farm, name string) *Farm {
+	if len(candidates) == 0 {
+		return nil
+	} else if len(candidates) == 1 {
+		return &candidates[0]
+	}
+
+	// other wise
+	for i := range candidates {
+		farm := &candidates[i]
+		if strings.EqualFold(farm.Name, name) {
+			return farm
+		}
+	}
+	// just choose the first one
+	return &candidates[0]
+}
+
+func migrate(ctx context.Context, v2 *directory.Farm, v3 *Farm) error {
+	// we finally know which farms we need to migrate from->to. we need
+	// to do the following:
+	// - first of all, find the usb stick, if it's not there, nothing we can do
+	// - stop provisiond + storaged
+	// - rewrite the usb stick with new v3 image with correct farm id
+	// - wipe the disks
+	// - sync
+	// - reboot
+	// - cross fingers
+
+	sticks, err := devices(ctx, IsUsb)
+	if err != nil {
+		return errors.Wrap(err, "failed to list usb sticks")
+	}
+
+	if len(sticks) == 0 {
+		// it's better if we error here so we don't proceed with update procedure
+		return fmt.Errorf("no usb sticks detected")
+	}
+
+	cl, err := zinit.New("/var/run/zinit.sock")
+	if err != nil {
+		return errors.Wrap(err, "failed to connect to zinit")
+	}
+	if err := cl.StopMultiple(2*time.Minute, "storaged", "provisiond"); err != nil {
+		return errors.Wrap(err, "Failed to stop services")
+	}
+
+	// if there are multiple usb sticks, we can't tell which one to use
+	// so we do all. good luck to the user
+	for _, usb := range sticks {
+		log.Info().Str("usb", usb.Path).Msg("rewriting usb stick")
+		if err := burn(ctx, v3.ID, &usb); err != nil {
+			return nil
+		}
+	}
+
+	syscall.Sync()
+
+	if err := wipe(ctx); err != nil {
+		return errors.Wrap(err, "failed to wipe disks")
+	}
+
+	syscall.Sync()
+
+	if err := crossFingers(); err != nil {
+		return errors.Wrap(err, "failed to reboot")
+	}
+	log.Info().Msg("I should never show up on screen")
+	return nil
+}
+
+func crossFingers() error {
+	f, err := os.OpenFile("/proc/sysrq-trigger", os.O_WRONLY, 0400)
```

Comment:
```go
syscall.Sync()`
err = syscall.Reboot(syscall.LINUX_REBOOT_CMD_RESTART)
```

Reply:
Yup, I started this with a "script" mindset.